docs/Migrating-To-ASK-SDK-v2-For-Java.rst

```diff
@@ -0,0 +1,379 @@
+Migration Guide for ASK SDK v2 for Java
+=======================================
+
+This guide shows how to use Alexa Skills Kit SDK v2 for Java to migrate
+existing skills developed with the SDK v1 to v2.
+
+Adding the ASK SDK v2 to Your Project
+-------------------------------------
+
+In the ``pom.xml``, from the SDK v1 version of the skill, replace the
+following:
+
+::
+
+   <dependency>
+         <groupId>com.amazon.alexa</groupId>
+         <artifactId>alexa-skills-kit</artifactId>
+         <version>1.8.1</version>
+         <scope>compile</scope>
+   </dependency>
+
+With the following new dependency. Note the change in ``artifactId`` and
+``version``.
+
+::
+
+   <dependency>
+         <groupId>com.amazon.alexa</groupId>
+         <artifactId>ask-sdk</artifactId>
+         <version>2.3.4</version>
+   </dependency>
+
+Request handlers
+----------------
+
+The SDK v2 introduces the concept of request handlers. Request handlers
+are discrete components that can handle one or more types of incoming
+requests. Handlers contain the canHandle method, which is responsible
+for deciding whether the handler should be invoked, and a handle method
+that contains the handling logic. This way, skill logic is structured
+and organized and not contained in one monolithic class.
+
+The following is from the Hello World sample written using the SDK v1.
+All the helper function implementations such as ``getWelcomeResponse``,
+``getAskResponse``, and ``getHelpResponse`` have been omitted because
+they are mentioned in a section on the response builder.
+
+::
```

Comment:
Let's use java codeblocks to have java code, for better code formatting in rst's 
``.. code:: java`` rather than just ``::``

Reply:
To make it consistent with other code blocks in current documentation, I would leave this as it is for now. But I can do another pull request that changes all the blocks in documentation to code blocks specific for Java.

Reply:
:+1: